src/parser/core/modules/Gauge/CounterGauge.ts

```diff
@@ -120,16 +124,36 @@ export class CounterGauge extends AbstractGauge {
 		})
 	}
 
+	/** @inheritdoc */
+	override generateResourceGraph() {
+		if (this.graphOptions == null) { return }
+
+		const {handle, color, label, collapse} = this.graphOptions
+		const graphData = {
+			label,
+			colour: color,
+			data: this.history.map(entry => {
+				return {time: entry.timestamp, current: entry.value, maximum: entry.maximum}
+			}),
+		}
+		if (handle != null) {
+			this.resourceGraphs.addDataGroup({...this.graphOptions, handle})
+			this.resourceGraphs.addData(handle, graphData)
+		} else {
+			this.resourceGraphs.addGauge(graphData, collapse)
+		}
+	}
+
 	/** @inheritdoc */
 	override generateDataset() {
-		// If there's no chart options, provide nothing
-		if (!this.chartOptions) {
+		// If there's no chart options, or if there are graph options, provide nothing (prefer graph)
+		if (this.chartOptions == null || this.graphOptions != null) {
```

Comment:
Why are these mutually exclusive? If a job module wants both, just show both. I realise it's duplicating information but the context of how it's presented is the important bit.

Reply:
No. One or the other.